crates/league-toolkit/src/core/meta/bin_tree.rs

```diff
@@ -0,0 +1,142 @@
+use std::{collections::HashMap, io};
+
+use super::{error::ParseError, BinTreeObject};
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug)]
+pub struct BinTree {
+    pub is_override: bool,
+    pub version: u32,
+
+    pub objects: HashMap<u32, BinTreeObject>,
+    /// List of other property bins we depend on.
+    ///
+    /// Property bins can depend on other property bins in a similar fashion to importing code libraries
+    pub dependencies: Vec<String>,
+
+    data_overrides: Vec<()>,
+}
+
+impl BinTree {
+    pub fn new(
+        objects: impl IntoIterator<Item = BinTreeObject>,
+        dependencies: impl IntoIterator<Item = String>,
+    ) -> Self {
+        Self {
+            version: 3,
+            is_override: false,
+            objects: objects
+                .into_iter()
+                .map(|o: BinTreeObject| (o.path_hash, o))
+                .collect(),
+            dependencies: dependencies.into_iter().collect(),
+            data_overrides: Vec::new(),
+        }
+    }
+}
+
+impl BinTree {
+    pub fn from_reader<R: io::Read>(reader: &mut R) -> Result<Self, ParseError> {
+        use crate::util::ReaderExt as _;
+        use byteorder::{ReadBytesExt as _, LE};
```

Comment:
Please fix these ugly imports, put them to the top of the file with the rest

Reply:
I have them nested so I can move them to a read.rs submodule later (once I add write support to this PR). Once they reach their final resting place I'll make it normal again